common/api.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_COMMON_API_HPP
+#define MENDER_COMMON_API_HPP
+
+#include <string>
+#include <vector>
+
+#include <common/common.hpp>
+#include <common/expected.hpp>
+#include <common/json.hpp>
+
+namespace mender {
+namespace common {
+namespace api {
+
+namespace common = mender::common;
+namespace expected = mender::common::expected;
+
+static inline expected::ExpectedString ErrorMsgFromErrorResponse(const vector<uint8_t> &response) {
+	auto ex_j = json::Load(StringFromByteVector(response));
+	if (!ex_j) {
+		return expected::unexpected(ex_j.error());
+	} else {
```

Comment:
Nit: Unnecessary `else` 
(I'm writing this a lot these days)

Reply:
Usually I'm for not using `else` because it introduces an unnecessary indentation for the code inside it. But here it reads well and there's just a different (**else**) `return` in it.